ruma-api-macros/src/api/request/outgoing.rs

```diff
@@ -131,23 +131,31 @@ impl Request {
             .collect();
 
         for auth in &metadata.authentication {
-            if auth.value == "AccessToken" {
-                let attrs = &auth.attrs;
-                header_kvs.extend(quote! {
+            let attrs = &auth.attrs;
+            let wants_access_token_auth = auth.value == "AccessToken";
+            header_kvs.extend(quote! {{
+                let access_token = match access_token {
+                    #ruma_api::SendAccessToken::IfRequired(access_token) => {
+                        if #wants_access_token_auth {
+                            Some(access_token)
+                        } else {
+                            None
+                        }
+                    },
```

Comment:
That is not what I meant. You're still emitting an `if else` expression even though the path that will be taken is know already at the point of the code being generated. I was thinking of something like

```suggestion
            let if_required_expr = if auth.value == "AccessToken" {
                quote! { Some(access_token) }
            } else {
                quote! { None }
            };
            header_kvs.extend(quote! {{
                let access_token = match access_token {
                    #ruma_api::SendAccessToken::IfRequired(access_token) => #if_required_expr,
```

Reply:
The `access_token` is not available outside of that `quote!`

Reply:
Ahhh, now I see what you mean, sorry

Reply:
It doesn't need to be.

Reply:
Identifiers in `quote!` are not hygienic in any way.

Reply:
Yeah, hadn't noticed that you quote!ed the Option as well, which is smart. Should read more carefully. Looks better now!